tests/unit/lifecycle/LifecycleTask.spec.js

```diff
@@ -353,6 +353,47 @@ describe('lifecycle task helper methods', () => {
 
             assert.deepStrictEqual(expected, res);
         });
+
+        it('should merge and sort arrays that include null version id ' +
+        'versions ', () => {
+            const versions = [
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373636323233323831343639393939393952473030312020363030',
+                    IsLatest: true,
+                    LastModified: '2018-04-04T23:16:46.000Z',
+                },
+                {
+                    Key: 'obj-1',
+                    VersionId: 'null',
+                    IsLatest: false,
+                    LastModified: '2018-04-04T23:16:46.000Z',
+                },
+            ];
+            const dms = [
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373636323233323831343639393939393952473030312020363035',
+                    IsLatest: false,
+                    LastModified: '2018-04-04T23:16:46.000Z',
```

Comment:
You may differentiate the last modified date from the null version's one to improve coverage (otherwise you're testing the "random" case which is mildly useful).

Also, you may add one more noncurrent delete marker that has an older timestamp than null version, the other with a more recent timestamp, and expect the null version to be stored in-between the two delete markers.